### DISCUSSION TITLE
How to get a overall list of all todos in all notes?

### DISCUSSION THREAD

**Discussion by cnshsliu** [nvim-telekasten/telekasten.nvim#211] [Q&A] [answered]
I may have may todos here and there, in different markdown notes.

I would like to be able to have a summarized view which contains all todos I have recorded so I have an overall view of what I have done/what not.

Very appreciate your help if you share your idea how to do it in telekasten, thanks in advance.

**Comment by lambtho12** [comment, marked as answer]
This is not possible at the moment. And a smiliar feature as already been discussed in #167. See my opinion on the matter: https://github.com/renerocksai/telekasten.nvim/pull/167#issuecomment-1261319437

What you could do is a custom mapping that will grep all notes and look for `^\-\s\[ \]` or `^\-\[x\]`.

**Reply by cnshsliu** [reply]
Could we have invisible (or automatically add) tags like `#todo ` for  `^\-\s\[ \]`  and  `#done` for `^\-\[x\]`, then we could leverage existing tag functions

**Reply by cnshsliu** [reply]
I made some change to the local codes to add '#TODO/DONE' tag automatically, so get a little bit easier to play todos with #TODO and #DONE tags

```
@@ -2776,18 +2776,23 @@ local function ToggleTodo(opts)
             and not vim.startswith(stripped, "- [")
         then
             repline = curline:gsub("%- ", "- [ ] ", 1)
+            repline = repline .. " #TODO"
         else
             if vim.startswith(stripped, "- [ ]") then
                 repline = curline:gsub("%- %[ %]", "- [x]", 1)
+                repline = repline:gsub("#TODO", "#DONE", 1)
             else
                 if vim.startswith(stripped, "- [x]") then
                     if opts.onlyTodo then
                         repline = curline:gsub("%- %[x%]", "- [ ]", 1)
+                        repline = repline:gsub("#DONE", "#TODO", 1)
                     else
                         repline = curline:gsub("%- %[x%]", "-", 1)
+                        repline = repline:gsub("#DONE", "", 1)
                     end
                 else
                     repline = curline:gsub("(%S)", "- [ ] %1", 1)
+                    repline = repline .. " #TODO"
                 end
             end
         end

```

**Comment by renerocksai** [comment]
We could also just use the existing search function, which is what I do.

cnshsliu ***@***.***> schrieb am Sa., 4. Feb. 2023, 01:32:

**Reply by cnshsliu** [reply]
Thanks a lot. yes, I tried search function as well and found it's convenient enough.
